### PR TITLE
Fix typo in  configsyncer service account template

### PR DESCRIPTION
**Pull request by Shubham-astro**
## Description

This PR intends to update the configsyncer SA helper template

## Related Issues

Do not merge this PR until this text is replaced with links to related issues.

Related astronomer/issues#6537

## Testing

Upon creation of custom SA configsyncer pod should be able to pick the correct configsyncer SA. 

## Merging

cherry pick to 0.36

### CODE REVIEW

**Comment by pgvishnuram** [comment]
do we have test cases covering this as well . if not can we add it pls

**Comment by Shubham-astro** [comment]
> do we have test cases covering this as well . if not can we add it pls

added

File: tests/chart_tests/test_byo_sa.py [hunk 1]
```diff
@@ -23,3 +23,29 @@ def test_role_created(self, kube_version):
 
         docs = [doc for doc in render_chart(kube_version=kube_version, values=values) if doc["kind"] in ["RoleBinding", "Role"]]
         assert not docs
+
+    def test_customSA(self, kube_version):
```
[the remark this answers is not in the document]
Comment (Shubham-astro) [reply tests/chart_tests/test_byo_sa.py:27]: done

**Review by pgvishnuram — APPROVED** [review]
lets cherry-pick to release-0.36 branch as well

**Pull request merged**